botnim/benchmark/evaluate_queries.py

```diff
@@ -0,0 +1,328 @@
+            else:
+                row = create_row_dict(
+                    group.loc[idx].to_dict(),
+                    metrics,
+                    was_retrieved=False
+                )
+                logger.info(f"Document {expected_doc} not found in results")
+            
+            rows.append(row)
+        
+        # Add unexpected retrieved documents
+        expected_docs = set(group['doc_filename'].tolist())
+        unexpected_docs = [(doc, score) for doc, score in doc_scores.items() if doc not in expected_docs]
+        unexpected_docs.sort(key=lambda x: x[1], reverse=True)
+        
+        for doc_name, score in unexpected_docs:
+            unexpected_row = {
+                'question_id': question_id,
+                'question_text': question_text,
+                'question_type': group['question_type'].iloc[0],
+                'doc_path': '',
+                'doc_filename': doc_name,
+                'comments/questions': '',
+                'is_expected': False,
+                'total_expected': total_expected,
+                **create_row_dict(
+                    {},
+                    metrics,
+                    was_retrieved=True,
+                    retrieved_rank=retrieved_docs.index(doc_name) + 1,
+                    retrieved_score=score
+                )
+            }
+            rows.append(unexpected_row)
+            logger.info(f"Added unexpected document: {doc_name} at rank {unexpected_row['retrieved_rank']} with score {score}")
+        
+        return rows
+        
+    except Exception as e:
+        logger.error(f"Error processing question {question_id}: {str(e)}")
+        return [
+            create_row_dict(
+                group.loc[idx].to_dict(),
+                QuestionMetrics(
+                    total_expected=total_expected,
+                    retrieved_count=0,
+                    last_expected_rank=0,
+                    recall_ratio=0,
+                    precision_ratio=0,
+                    f1_score=0,
+                    adjusted_f1_score=0
+                ),
+                was_retrieved=False
+            )
+            for idx in group.index
+        ]
+
+def evaluate_queries(
+    csv_path: str,
+    store_id: str,
+    adjusted_f1_limit: int = 7,
+    max_results: int = MAX_RESULTS
+) -> pd.DataFrame:
+    """
+    Evaluate queries from CSV file and mark which expected documents were retrieved
+    
+    Args:
+        csv_path (str): Path to the CSV file containing questions and expected documents
+        store_id (str): The vector store ID to query against
+        adjusted_f1_limit (int): Number of documents to consider for adjusted F1 score calculation
+        max_results (int): Maximum number of results to retrieve per query
+        
+    Returns:
+        pd.DataFrame: Updated DataFrame with retrieval information
+    """
+    # Read the CSV file
+    df = pd.read_csv(csv_path)
+    
+    # Parse store_id
+    parse_store_id(store_id)
```

Comment:
Why do you need this function?
If it's just for logging, then it's better to add a log print before you encode the store_id, no?